src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -235,6 +245,16 @@ private FunctionDescr Scan()
                     {
                         throw new InvalidOperationException($"Cancellation token must be the last argument.");
                     }
+                    else if (parameters[i].ParameterType == typeof(Func<Task<BooleanValue>>))
+                    {
+                        info.LamdaParamMask = info.LamdaParamMask | BigInteger.One << i;
+                        paramTypes.Add(FormulaType.Boolean);
+                    }
+                    else if (parameters[i].ParameterType == typeof(Func<Task<FormulaValue>>))
```

Comment:
> == typeof(Func<Task<FormulaValue>> [](http://example.com/codeflow?start=56&length=35)

This won't catch Func<Task<NumberValue>>  or other derived cases. 

You have to use some reflection APIs. 
Se Utility.TryGetElementType